### PR TITLE
Add extra column to batches

### PR DESCRIPTION
**Pull request by xrigau**
This PR is very similar to #88 - only difference is it adds the EXTRA_DATA column to the Batches.

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/137/
Test FAILed.

File: demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/batches/BatchDownloadsActivity.java [hunk 1]
```diff
@@ -83,6 +83,7 @@ private void enqueueBatch() {
                 .withDescription("Goatees galore")
                 .withBigPictureUrl(BEARD_IMAGE)
                 .withVisibility(NotificationVisibility.ACTIVE_OR_COMPLETE)
+                .withExtraData("An extra beard.")
```
Comment (Electryc) [inline demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/batches/BatchDownloadsActivity.java]: :+1: I like

File: library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java [hunk 2]
```diff
@@ -125,6 +126,17 @@ public Builder withStatusFilter(@Status int statusFlags) {
             return this;
         }
 
+        /**
+         * Filter by extra data.
+         *
+         * @return {@link BatchQuery.Builder}
+         */
+        public Builder withExtraData(String extraData) {
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java]: No test for this :panda_face: https://github.com/novoda/download-manager/blob/master/library/src/test/java/com/novoda/downloadmanager/lib/BatchQueryTest.java

File: library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java [hunk 3]
```diff
@@ -125,6 +126,17 @@ public Builder withStatusFilter(@Status int statusFlags) {
...
+            this.criteriaExtraDataBuilder = new Criteria.Builder();
+            criteriaExtraDataBuilder.withSelection(DownloadContract.Batches.COLUMN_EXTRA_DATA, Criteria.Wildcard.EQUALS);
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java]: :x: missing `withArgument(extraData)`.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/139/
Test PASSed.

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/140/
Test PASSed.

File: library/src/test/java/com/novoda/downloadmanager/lib/BatchQueryTest.java [hunk 4]
```diff
@@ -26,6 +26,15 @@ public void givenTheQueryIsWithIdWhenItIsBuiltThenTheSelectionAndArgumentsAreCor
     }
 
     @Test
+    public void givenTheQueryIsWithExtraDataWhenItIsBuiltThenTheSelectionAndArgumentsAreCorrect() {
```
Comment (Electryc) [inline library/src/test/java/com/novoda/downloadmanager/lib/BatchQueryTest.java]: :+1:

**Pull request merged**